mdata/cache/ccache_metric.go

```diff
@@ -101,15 +101,22 @@ func (mc *CCacheMetric) AddRange(prev uint32, itergens []chunk.IterGen) {
 		// if previous chunk has not been passed we try to be smart and figure it out.
 		// this is common in a scenario where a metric continuously gets queried
 		// for a range that starts less than one chunkspan before now().
-		if prev == 0 {
-			res, ok := mc.seekDesc(ts - 1)
-			if ok {
+		res, ok := mc.seekDesc(ts - 1)
+		if ok {
+			if prev == 0 {
 				prev = res
+			} else if prev != res {
```

Comment:
nice, i think if this condition is true, then this could lead to the corruption case which looks like what you posted in your gist:
```
2018/07/19 15:56:46 [W] CCacheMetric debugMetric: ts 1531249200; prev 1531245600; next 1531252800
2018/07/19 15:56:46 [W] CCacheMetric debugMetric: ts 1531252800; prev 1531249200; next 0  
2018/07/19 15:56:46 [W] CCacheMetric debugMetric: ts 1531350000; prev 1531249200; next 1531353600 
```
Did you ever see such warnings in a prod environment being emitted?